src/core/lib/event_engine/posix_engine/posix_engine.cc

```diff
@@ -92,6 +95,18 @@ void PosixEventEngine::Run(EventEngine::Closure* closure) {
   executor_.Run(closure);
 }
 
+absl::Status PosixEventEngine::WaitForPendingTasks(Duration timeout) {
```

Comment:
Is this function equivalent to (sans timeout):
```
executor_.PrepareFork();
executor_.PostFork();
```
in its observable behavior?

(if yes: should we formalize that pair and use it?)

Reply:
To the application, yes, given that there are no closures pending when `PrepareFork` returns.

Mechanically (and observable to system monitoring and whatnot), that combination would start and stop N threads, which is unnecessary.

Reply:
Yeah... I'm trying to judge the trade off of adding another shutdown path versus refining an existing one. Yes one path would shutdown N threads: my assumption is we wouldn't care.

Reply:
That's a fair argument. Thinking through caveats, it could get weird if a fork occurs when we're attempting to drain the queue using the same approaches. We'll need some special handling for those events overlapping if we want to reuse the code that starts & stops threads. As the PR exists now, I think it would work fine without that extra overhead. But the reuse is doable. WDYT?